### PR TITLE
refactor hgraph_params to solve static value leak

### PR DESCRIPTION
**Pull request by LHT129**
fixed: #222

### CODE REVIEW

File: src/index/hgraph_zparameters.cpp [hunk 1]
```diff
@@ -23,14 +23,59 @@
...
-     {HGRAPH_BASE_QUANTIZATION_TYPE, {HGRAPH_BASE_CODES_KEY, QUANTIZATION_TYPE_KEY}},
-     {HGRAPH_GRAPH_MAX_DEGREE, {HGRAPH_GRAPH_KEY, GRAPH_PARAMS_KEY, GRAPH_PARAM_MAX_DEGREE}},
-     {HGRAPH_BUILD_EF_CONSTRUCTION, {BUILD_PARAMS_KEY, BUILD_EF_CONSTRUCTION}}};
+static const std::unordered_map<std::string, std::vector<std::string>> EXTERNAL_MAPPING = {
+    {HGRAPH_USE_REORDER, {HGRAPH_USE_REORDER_KEY}},
+    {HGRAPH_BASE_QUANTIZATION_TYPE, {HGRAPH_BASE_CODES_KEY, QUANTIZATION_TYPE_KEY}},
+    {HGRAPH_GRAPH_MAX_DEGREE, {HGRAPH_GRAPH_KEY, GRAPH_PARAMS_KEY, GRAPH_PARAM_MAX_DEGREE}},
+    {HGRAPH_BUILD_EF_CONSTRUCTION, {BUILD_PARAMS_KEY, BUILD_EF_CONSTRUCTION}}};
+
+static const std::string DEFAULT_HGRAPH_PARAMS_STR =
```
Comment (jiaweizone) [inline src/index/hgraph_zparameters.cpp]: DEFAULT_HGRAPH_PARAMS_STR -> HGRAPH_PARAMS_TEMPLATE ?

File: src/index/hgraph_zparameters.cpp [hunk 2]
```diff
@@ -23,14 +23,59 @@
...
+            "{BUILD_EF_CONSTRUCTION}": 400,
+            "{BUILD_THREAD_COUNT}": 100
+        }
+    })";
 
 HGraphParameters::HGraphParameters(JsonType& hgraph_param, const IndexCommonParam& common_param)
-    : common_param_(common_param) {
+    : common_param_(common_param),
+      default_hgraph_params_(format_map(DEFAULT_HGRAPH_PARAMS_STR, DEFAULT_MAP)),
+      hgraph_param_mapping_(EXTERNAL_MAPPING) {
```
Comment (jiaweizone) [inline src/index/hgraph_zparameters.cpp]: Why need an `EXTERNAL_MAPPING` object copy ?

Comment (LHT129) [reply src/index/hgraph_zparameters.cpp]: remove the value now.

File: src/index/hgraph_zparameters.cpp (hunk 1 above)
Comment (LHT129) [reply src/index/hgraph_zparameters.cpp]: done

**Review by jiaweizone — APPROVED** [review]
LGTM

**Review by jiaweizone — APPROVED** [review]
LGTM

**Review by inabao — APPROVED** [review]
LGTM

**Pull request merged**